project/src/main/java/com/Main.java

```diff
@@ -21,10 +28,79 @@ public class Main {
+        Motorbike motorbike4 = MOTORBIKE_SERVICE.create("S 3600 RR", Manufacturer.BMW, BigDecimal.valueOf(300), 60.0, 5);
+        Airplane airplane = AIRPLANE_SERVICE.create("Боинг-777", Manufacturer.BMW, BigDecimal.valueOf(1_000_000), 1000, 1);
+        List<Auto> autoList = AUTO_SERVICE.create(1);
+        List<Vehicle> vehicleList = new LinkedList<>();
+        vehicleList.add(motorbike1);
+        vehicleList.add(airplane);
+        vehicleList.add(motorbike2);
+        vehicleList.add(autoList.get(0));
+        vehicleList.add(motorbike3);
+        vehicleList.add(motorbike4);
+        Parking<Vehicle> parking = new Parking<>();
+        parking.add(motorbike1);
+        parking.add(airplane);
+        parking.add(autoList.get(0));
+        LOGGER.info(parking.print());
+        StringJoiner stringJoiner = new StringJoiner("--");
+        for (Vehicle vehicle : parking) {
+            stringJoiner.add(vehicle.getClass().getSimpleName());
+        }
+        LOGGER.info(stringJoiner.toString());
+        try {
+            MOTORBIKE_SERVICE.print((Motorbike) parking.search(parking.getRestylingNumber(motorbike1)));
+        } catch (NoSuchElementException exception) {
+            LOGGER.error(exception.getMessage());
+        }
+        try {
+            MOTORBIKE_SERVICE.print((Motorbike) parking.search("0"));
+        } catch (NoSuchElementException exception) {
+            LOGGER.error(exception.getMessage());
+        }
+        parking.set(parking.getRestylingNumber(motorbike1), airplane);
+        LOGGER.info(parking.print());
+        parking.set(parking.getRestylingNumber(airplane), autoList.get(0));
+        LOGGER.info(parking.print());
+        parking.set(parking.getRestylingNumber(autoList.get(0)), motorbike1);
+        LOGGER.info(parking.print());
+        parking.remove(parking.getRestylingNumber(motorbike1));
+        LOGGER.info(parking.print());
+        LOGGER.info("Number of restyling: {}", parking.getSize());
+        LOGGER.info("Date of parking of the first vehicle {}", parking.getHeadTime());
+        LOGGER.info("Date of parking of the last vehicle {}", parking.getTailTime());
+
+        //Comparator
+        StringBuilder stringBuilder = new StringBuilder();
+        LOGGER.info("List before sorting: ");
+        for (Vehicle vehicle : vehicleList) {
+            stringBuilder
+                    .append(vehicle.getPrice())
+                    .append(" ")
+                    .append(vehicle.getClass().getSimpleName())
+                    .append(" ")
+                    .append(vehicle.getCount())
+                    .append(" ");
+        }
+        LOGGER.info(stringBuilder.toString());
+
+        Vehicle.VehicleComparator vehicleComparator = new Vehicle.VehicleComparator();
+        vehicleComparator.thenComparing(vehicle -> vehicle.getClass().getSimpleName()).thenComparing(Vehicle::getCount);
```

Comment:
```suggestion
        final Comparator<Vehicle> comparator = Comparator.comparing(Vehicle::getPrice)
                .reversed()
                .thenComparing(vehicle -> vehicle.getClass().getSimpleName())
                .thenComparing(Vehicle::getCount);
```
Сортировка по цене не верная. 
Я бы тогда делал все компараторы на анонимных классах